src/api.rs

```diff
@@ -234,6 +234,21 @@ impl ThreadPool {
             job_a.into_result()
         }
     }
+
+    /// Returns the number of threads in the thread pool.
+    pub fn num_threads(&self) -> usize {
+        self.registry.num_threads()
+    }
+
+    /// Returns the index of the current thread in the thread pool. Panics if
+    /// the current thread is not in this thread pool.
```

Comment:
(That said, I think I would have expected this to just be a free function. And I'd probably prefer for it to just return `Option<usize>` rather than panic.)